### PR TITLE
Bump hf libraries versions

### PR DESCRIPTION
**Pull request by JingyaHuang**
To solve the tokenizers vs huggingface_hub version conflicts. @dacorvo

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 
 INSTALL_REQUIRES = [
-    "transformers == 4.35.0",
+    "transformers == 4.36.2",
     "accelerate == 0.23.0",
     "optimum >= 1.14.0",
-    "huggingface_hub >= 0.14.0",
+    "huggingface_hub >= 0.19.4",
```
Comment (michaelbenayoun) [inline setup.py:19]:
```suggestion
    "huggingface_hub >= 0.20.1",
```
I think it'll be needed for us to use the `huggingface_hub.get_token()` function, which is the recommended way to ge the value of the token from now on.

File: .github/workflows/test_inf2.yml [hunk 2]
```diff
@@ -38,20 +38,20 @@ jobs:
       - name: Run CLI tests
         run: |
           source aws_neuron_venv_pytorch/bin/activate
-          HF_TOKEN_OPTIMUM_NEURON_CI=${{ secrets.HF_TOKEN_OPTIMUM_NEURON_CI }} pytest -m is_inferentia_test tests/cli
+          HF_TOKEN=${{ secrets.HF_TOKEN_OPTIMUM_NEURON_CI }} pytest -m is_inferentia_test tests/cli
       - name: Run exporters tests
         run: |
           source aws_neuron_venv_pytorch/bin/activate
-          HF_TOKEN_OPTIMUM_NEURON_CI=${{ secrets.HF_TOKEN_OPTIMUM_NEURON_CI }} pytest -m is_inferentia_test tests/exporters
```
Comment (michaelbenayoun) [inline .github/workflows/test_inf2.yml:45]: Is this environment variable only used in the two workflows you changed?

Comment (JingyaHuang) [reply .github/workflows/test_inf2.yml:45]: I saw Trainium tests are using it as well, do you want me to change trainium CIs as well?

Comment (michaelbenayoun) [reply .github/workflows/test_inf2.yml:45]: Let's do it in another PR.

**Review by michaelbenayoun — APPROVED** [review]
The CI for training do not pass, I think that's ok. We can iterate later. I think that #279 will solve many of these issues.

**Pull request merged**